tizen-api/include/nnstreamer-api-simpleshot.h

```diff
@@ -0,0 +1,149 @@
+ *         We intend to start ACR after ACR for nnstreamer-main is completed.
+ */
+
+#ifndef __TIZEN_NNSTREAMER_API_SIMPLESHOT_H__
+#define __TIZEN_NNSTREAMER_API_SIMPLESHOT_H__
+
+#include <tizen_error.h>
+#include <nnstreamer/tensor_typedef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+/**
+ * @addtogroup CAPI_ML_NNSTREAMER_SIMPLESHOT_MODULE
+ * @{
+ */
+
+
+/**
+ * @brief A handle of a simpleshot instance
+ * @since_tizen 5.5
+ */
+typedef void *ml_simpleshot_model_h;
+
+/* We can handle up to 16 tensors for an input or output frame */
+#define ML_MAX_TENSORS	(16)
+/**
+ * @brief An instance of input or output frames. GstTensorsInfo is the metadata
+ */
+typedef struct {
+  void *tensor[ML_MAX_TENSORS]; /**< Set NULL for unused tensors. If there is a single
+                                     tensor in a frame, every tensor[i] is NULL except
+                                     for tensor[0]. */
+} tensor_data;
+
+/**
+ * @brief Types of NNFWs.
+ */
+typedef enum {
+  ML_NNFW_CUSTOM_FILTER = 0,
+  ML_NNFW_TENSORFLOW_LITE,
+  ML_NNFW_TENSORFLOW,
+} ml_model_nnfw;
+
+/**
+ * @brief Types of NNFWs. Note that if the affinity (nnn) is not supported by the driver or hardware, it is ignored.
+ */
+typedef enum {
+  ML_NNFW_HW_DO_NOT_CARE = 0,
+  ML_NNFW_HW_AUTO = 1,
+  ML_NNFW_HW_CPU = 0x1000, /**< 0x1000: any CPU. (nnn>0) 0x1nnn: CPU # nnn-1 */
+  ML_NNFW_HW_GPU = 0x2000, /**< 0x2000: any GPU. (nnn>0) 0x2nnn: GPU # nnn-1 */
+  ML_NNFW_HW_NPU = 0x3000, /**< 0x3000: any NPU. (nnn>0) 0x3nnn: NPU # nnn-1 */
+} ml_model_hw;
+
+/*************
+ * MAIN FUNC *
+ *************/
+/**
+ * @brief Open an ML model, detecting NNFW automatically based on the filepath extension.
+ * @param[in] inputtype This is required if the given model has flexible input
+ *                     dimension, where the input dimension MUST be given before
+ *                     executing the model. However, once it's given, the input
+ *                     dimension cannot be changed for the given model handle.
+ *                     Is is required by a few of custom filters of nnstreamer.
+ *                     You may set NULL if it's not required.
+ */
+int ml_model_open (const char *model_path, ml_simpleshot_model_h *model,
+    const GstTensorsInfo *inputtype);
+
+/**
+ * @brief ml_model_open with nnfw manually designated.
+ */
+int ml_model_open2 (const char *model_path, ml_model_nnfw nnfw,
+    ml_simpleshot_model_h *model, const GstTensorsInfo *inputtype);
+
+/**
+ * @brief ml_model_open with nnfw and its hardware backend manually designated.
+ */
+int ml_model_open3 (const char *model_path, ml_model_nnfw nnfw,
+    ml_model_hw hw, ml_simpleshot_model_h *model,
+    const GstTensorsInfo *inputtype);
+
+int ml_model_close (ml_simpleshot_model_h model);
+
+int ml_model_inference (ml_simpleshot_model_h model, const tensor_data *input,
+    tensor_data *output);
+
+tensor_data * ml_model_inference_alloc (ml_simpleshot_model_h model,
```

Comment:
what is the purpose of this function?

Reply:
It is to let the inference-API allocate output buffer for the user. (allocate_in_invoke = TRUE)

On the other hand, with ```ml_model_inference```, the user (application writer) must allocate output buffer for the API.